src/components/ClusterTemplateQuotas/NewQuotaDialog/useCreateQuota.ts

```diff
@@ -0,0 +1,88 @@
+import { useK8sModel, k8sCreate, K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  clusterTemplateQuotaGVK,
+  clusterTemplatesRoleRef,
+  namespaceGVK,
+  RBAC_API_GROUP,
+  roleBindingGVK,
+} from '../../../constants';
+import { Quota, RoleBinding, Subject } from '../../../types';
+import * as React from 'react';
+import { getApiVersion } from '../../../utils/k8s';
+import { NewQuotaFormikValues } from '../types';
+import { useNamespaceContext } from '../../../contexts/NamespaceContextProvider';
+
+const getQuota = (values: NewQuotaFormikValues): Quota => {
+  return {
+    apiVersion: getApiVersion(clusterTemplateQuotaGVK),
+    kind: clusterTemplateQuotaGVK.kind,
+    metadata: {
+      name: values.name,
+      namespace: values.namespace,
+    },
+    spec: {
+      budget: values.hasBudget ? values.budget : undefined,
+      allowedTemplates: [],
+    },
+  };
+};
+
+const getNamespace = (name: string): K8sResourceCommon => {
+  return {
+    apiVersion: namespaceGVK.version,
+    kind: namespaceGVK.kind,
+    metadata: {
+      name,
+    },
+  };
+};
+
+const getSubjects = (names: string[], kind: Subject['kind']): Subject[] =>
+  names.map((name) => {
+    return {
+      kind,
+      apiGroup: RBAC_API_GROUP,
+      name: name,
+    };
+  });
+
+const getRoleBinding = (namespace: string, users: string[], groups: string[]): RoleBinding => ({
+  metadata: {
+    generateName: 'cluster-templates-rb-',
+    namespace,
+  },
+  subjects: [...getSubjects(users, 'User'), ...getSubjects(groups, 'Group')],
+  roleRef: clusterTemplatesRoleRef,
+});
+
+const useCreateQuota = (): {
+  create: (values: NewQuotaFormikValues) => Promise<Quota>;
+} => {
+  const [quotaModel] = useK8sModel(clusterTemplateQuotaGVK);
+  const [roleBindingModel] = useK8sModel(roleBindingGVK);
+  const [namespaceModel] = useK8sModel(namespaceGVK);
```

Comment:
we should make sure loading is handled properly

Reply:
Handling the loading and error state is complicated
What do you think of simply having these models hardcoded in the frontend. That's what I did for NodeHealthChecks.
The required properties are:
group, version, kind, plural, labelPlural and abbr
group, version and kind are anyhow hard coded already
I think having plural, labelPlural and abbr hardcoded is not the worst thing.
WDYT?